src/commands/runDraftTool/runDraftWorkflow.ts

```diff
@@ -0,0 +1,467 @@
+   }
+}
+
+class PromptACRSelection extends AzureWizardPromptStep<WizardContext> {
+   constructor(private az: AzApi) {
+      super();
+   }
+
+   public async prompt(wizardContext: WizardContext): Promise<void> {
+      if (wizardContext.clusterSubscription === undefined) {
+         throw Error('ACR Subscription is undefined');
+      }
+      if (wizardContext.acrResourceGroup === undefined) {
+         throw Error('ACR Resource Group is undefined');
+      }
+
+      const registries = getAysncResult(
+         this.az.listContainerRegistries(
+            wizardContext.clusterSubscription,
+            wizardContext.acrResourceGroup
+         )
+      );
+
+      const registryToItem = (r: RegistryItem) => ({
+         label: r.registry.name || ''
+      });
+
+      const registryPick = await wizardContext.ui.showQuickPick(
+         sort(getAsyncOptions(registries, registryToItem)),
+         {
+            ignoreFocusOut,
+            stepName: 'Registry',
+            placeHolder: 'Registry',
+            noPicksMessage: 'No Registries found'
+         }
+      );
+
+      wizardContext.registry = (await registries).find(
+         (r) => registryToItem(r).label === registryPick.label
+      );
+   }
+
+   public shouldPrompt(wizardContext: WizardContext): boolean {
+      return true;
+   }
+}
+
+class PromptACRRegistrySelection extends AzureWizardPromptStep<WizardContext> {
+   constructor(private az: AzApi) {
+      super();
+   }
+
+   public async prompt(wizardContext: WizardContext): Promise<void> {
+      if (wizardContext.clusterSubscription === undefined) {
+         throw Error('ACR Subscription is undefined');
+      }
+      if (wizardContext.registry === undefined) {
+         throw Error('ACR Registry is undefined');
+      }
+
+      const repositories = getAysncResult(
+         this.az.listRegistryRepositories(
+            wizardContext.clusterSubscription,
+            wizardContext.registry
+         )
+      );
+      const repositoryToItem = (r: RepositoryItem) => ({
+         label: r.repositoryName
+      });
+
+      const repositoryPick = await wizardContext.ui.showQuickPick(
+         sort(getAsyncOptions(repositories, repositoryToItem)),
+         {
+            ignoreFocusOut,
+            stepName: 'Repository',
+            placeHolder: 'Repository',
+            noPicksMessage: 'No Repositories found'
+         }
+      );
+
+      wizardContext.acrRepository = (await repositories).find(
+         (r) => repositoryToItem(r).label === repositoryPick.label
+      );
+   }
+
+   public shouldPrompt(wizardContext: WizardContext): boolean {
+      return true;
+   }
+}
+
+class PromptGitHubBranchSelection extends AzureWizardPromptStep<WizardContext> {
+   public async prompt(wizardContext: WizardContext): Promise<void> {
+      if (wizardContext.destination === undefined) {
+         throw Error('Git Directory is Undefined');
+      }
+      const branches = await getBranches(wizardContext.destination);
```

Comment:
We should be filtering for only remote branches. The selected branch is the branch on github that will trigger deploys on push. Right now they will have the option to select local only branches.

Reply:
This information should already be in the branch objects so you can just filter here.

Reply:
good call, swapped to filtering out local branches and removing remote definition from labels.